lesson2/job1/api.py

```diff
@@ -0,0 +1,59 @@
+import requests
+import os
+
+API_URL = "https://fake-api-vycpfa6oca-uc.a.run.app/"
+
+
+class ApiException(Exception):
+    pass
+
+
+def is_blank(string):
+    if string:
+        return False
+    return True
+
+
+def get_sales(date: str, path: str):
+    token = os.getenv('AUTH_TOKEN')
+    if is_blank(token):
+        raise ApiException("Auth Token not set. Exiting...")
+
+    headers = {"Authorization": token}
+    basedir = os.path.expanduser(
+        '~/Projects/robot-dreams-dataeng/lesson2/data'
+        )
+    storing_path = os.path.join(basedir, path.lstrip('/'))
+
+    if not os.path.exists(storing_path):
+        try:
+            os.makedirs(storing_path)
+        except Exception as e:
```

Comment:
When you use a general exception, it's called [Error Hiding Antipattern](https://en.wikipedia.org/wiki/Error_hiding) because it may hide any error inside `try` clause